sim/src/sim_client.h

```diff
@@ -149,7 +149,7 @@ namespace crimson {
 
     public:
 
-      SimulatedClient(ClientId _id,
+      SimulatedClient(const ClientId& _id,
```

Comment:
If these "Id" types are actually just ints, you're probably better off passing them as const values rather than const references.

Reply:
Thanks for pointing it out, but ClientId is [not a primitive](https://github.com/ceph/ceph/blob/2739b6f1c52a45aa22a62313d140d769be532874/src/librbd/watcher/Types.h#L19-L47) type.

Reply:
No problem-- the other thing I should point out after I hem and haw with myself is that with const& in tricker situations (like here, where I wrongly assumed it was as per the definition at "https://github.com/Rubonnek/dmclock/blob/41d97df5493b23bcbb8c3aaf53934231c82e48e4/sim/src/sim_recs.h"), any errors will be on the side of good. ;-)

Reply:
Welp, now I think you are right. I first saw the ClientId when I scanned the whole ceph repository with `cppcheck` which included this dmclock repository. I thought ClientId was also defined as a struct here, but it's not, it's just what you pointed out.

I also didn't notice the `inconclusive` flag from `cppcheck`. That usually brings up false positives and this change is included with that flag.

Thanks for the review! I'll drop these changes, and the other inconclusive ones.

Reply:
Glad to be of help!